src/sql/src/plan/statement/ddl.rs

```diff
@@ -3044,6 +3066,19 @@ fn kafka_sink_builder(
         None => bail_unsupported!("sink without format"),
     };
 
+    let mut topic_config = topic_config.unwrap_or_default();
+    // If the user did not provide a cleanup policy for the topic, default to infinite
+    // retention.
+    if !topic_config.contains_key("cleanup.policy") {
+        topic_config.insert("cleanup.policy".to_string(), "delete".to_string());
+        if topic_config.contains_key("retention.bytes") || topic_config.contains_key("retention.ms")
+        {
+            sql_bail!("Cannot specify retention.bytes or retention.ms without cleanup.policy");
+        }
+        topic_config.insert("retention.bytes".to_string(), "-1".to_string());
+        topic_config.insert("retention.ms".to_string(), "-1".to_string());
+    }
```

Comment:
I know the old code did this, but it is at odds with our documentation: https://materialize.com/docs/sql/create-sink/kafka/#automatic-topic-creation

> If the specified Kafka topic does not exist, Materialize will attempt to create it using the broker’s default number of partitions, default replication factor, default compaction policy, and default retention policy.

I'm inclined to make the code match the docs here and just _not_ set anything about `cleanup.policy` or `retention.{bytes,ms}` by default.